src/server/pkg/storage/chunk/option.go

```diff
@@ -19,10 +19,21 @@ func WithGarbageCollection(gcClient gc.Client) StorageOption {
 	}
 }
 
+// WithMaxConcurrentObjects sets the maximum number of object writers (upload)
+// and readers (download) that can be open at a time.
+func WithMaxConcurrentObjects(maxUpload, maxDownload int) StorageOption {
```

Comment:
The fact that the parameters for this function are upload, download and the internal call's are download, upload is bothering me lol. Would you mind making those consistent?